### ISSUE TITLE
Having multiple instances of lock prevents authenticated event

### ISSUE THREAD

**Issue by Abdisalan** [auth0/lock#1285]
Using:
auth0-lock@v11.3.1
MacOS Sierra 10.12.6 (16G1212)
Chrome Version 64.0.3282.186 (Official Build) (64-bit)

This bug can be recreated using create-react-app and adding this auth.js file and App.jsx

```javascript
// auth.js
import Auth0Lock from 'auth0-lock';

const cid = '<clientId>';
const domain = '<domain>';

const options = {
    auth: {
        redirectUrl: 'http://localhost:3000',
        responseType: 'token',
    }
};

const lock1 = new Auth0Lock(cid, domain, options);

export const initLock = (state) => {
    options.auth.params = {state: state};
    return new Auth0Lock(cid, domain, options);
};
```

```javascript
// App.jsx
import React, { Component } from 'react';
import './App.css';

import {initLock} from './auth.js';

class App extends Component {
  constructor(props) {
    super(props);

    this.lock2 = initLock('/');
    this.lock2.on('authenticated', (authResult) => {
      console.log(authResult);
    })

    this.showLock = this.showLock.bind(this);
  }

  showLock() {
    this.lock2.show();
  }

  render() {
    return (
      <div>
        <button onClick={this.showLock}>Login</button>
      </div>
    );
  }
}

export default App;
```

Solution is currently commenting out `lock1` in `auth.js`

It took me a while to find this issue because this code works with all versions of `lockv10` and I use a few locks in our app.

**Comment by luisrudge** [comment]
Hi, v10 also doesn't emit the events in the second instance in my tests:

```html
<!DOCTYPE html>
<html>
<title>Dev</title>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0" />

<body>
  <script src="https://cdn.auth0.com/js/lock/10.24/lock.min.js"></script>
  <script>
    const clientId = "BWDP9XS89CJq1w6Nzq7iFOHsTh6ChS2b";
    const domain = "brucke.auth0.com";
    const defaultOptions = {
      oidcConformant: true,
      allowShowPassword: true,
      usernameStyle: 'email',
      defaultDatabaseConnection: 'acme',
      prefill: {
        email: 'johnfoo@gmail.com'
      },
      passwordlessMethod: 'code',
      allowedConnections: ['email', 'acme', 'twitter']
    };
    var lock = new Auth0Lock(clientId, domain, defaultOptions);
    var lock2 = new Auth0Lock(clientId, domain, defaultOptions);
    //if you use lock.on, it works
    lock2.on('authenticated', function (auth) {
      console.log(auth);
    });
    lock.show();
  </script>
</body>

</html>
```

Your best option here is to set `autoParseHash: false` in the instances that are not going to handle the authenticated event (`lock1` in your case). This way, when your `lock2` instance is created, it will parse the hash and emit the event.

**Comment by Abdisalan** [comment]
Thanks for the reply! That's odd, because that setup works for me on `auth0-lock@10.24`, just not with lock v11 since `auth0-lock@11.0.0`. Here's my react version.

```javascript
// App.js
import React, { Component } from 'react';

import Auth0Lock from 'auth0-lock';

const cid = '<cid>';
const domain = '<domain>';

const options = {
    auth: {
        redirectUrl: 'http://localhost:3000',
        responseType: 'token',
    }
};

class App extends Component {
  constructor(props) {
    super(props);

    this.lock1 = new Auth0Lock(cid, domain, options);

    this.lock2 = new Auth0Lock(cid, domain, options);
    this.lock2.on('authenticated', (authResult) => {
      console.log(authResult);
    })

    this.showLock2 = this.showLock2.bind(this);
  }

  showLock2() {
    this.lock2.show();
  }

  render() {
    return (
      <div>
        <button onClick={this.showLock2}>Login</button>
      </div>
    );
  }
}

export default App;
```

Also sorry I didn't clarify why I was using two locks; My use case is to sign the user in with one lock, then while logged in allow them to sign into their linkedin with another lock to connect the accounts. Each lock in this case would need to have a different options object. This is because there are settings I can't override when I use `lock.show(options)` like `auth.redirectUrl`.


Here's a basic version of the code I used in v10 that works but breaks on v11. When using v10, both locks receive the authenticated event. When using v11, neither receive it.

```javascript
// App.js
import React, { Component } from 'react';

import Auth0Lock from 'auth0-lock';

const cid = '<cid>';
const domain = '<domain>';

const options = {
    auth: {
        redirectUrl: 'http://localhost:3000',
        responseType: 'token',
    }
};

const options2 = {
  // in my code this uses a different redirect URI but for simplicity they are the same here
  auth: {
    redirectUrl: 'http://localhost:3000',
    responseType: 'token',
  },
  allowedConnections: ['linkedin']
}

class App extends Component {
  constructor(props) {
    super(props);

    this.lock1 = new Auth0Lock(cid, domain, options);
    this.lock1.on('authenticated', (authResult) => {
      // login
      console.log(authResult)
    });

    this.lock2 = new Auth0Lock(cid, domain, options2);
    this.lock2.on('authenticated', (authResult) => {
      // link social account
      console.log(authResult);
    })


    this.showLock = this.showLock.bind(this);
    this.showLock2 = this.showLock2.bind(this);
  }

  showLock() {
    this.lock1.show();
  }

  showLock2() {
    this.lock2.show();
  }

  render() {
    return (
      <div>
        <button onClick={this.showLock}>Login</button>
        <button onClick={this.showLock2}>Connect Account</button>
      </div>
    );
  }
}

export default App;
```

Is there a new pattern recommended by auth0 I should use for connecting accounts in v11? Or Am I just using locks wrong?

Thank you for taking that time to read through this really long post.

**Comment by luisrudge** [comment]
You can use multiple locks, but only one of them (by default, the first created) will be able to automatically parse the hash by default. We have a lot of docs about account linking that you can check: https://auth0.com/docs/link-accounts. The [SPA sample](https://github.com/auth0-samples/auth0-link-accounts-sample/blob/master/SPA) looks just like yours (multiple lock instances), but you can see [here](https://github.com/auth0-samples/auth0-link-accounts-sample/blob/master/SPA/app.js#L336) that only one instance is listening to the authenticated event.

**Comment by Abdisalan** [comment]
Well, this is getting weird.. I cloned that repo, put in my clientId and domain, was able to give the app access, was able to login and have my profile json show. BUT when I connected my linkedin, my account wasn't linked. It was a successful authentication because it created an account on my dashboard. No error messages in the console as well.

I think there must be something wrong in my client for this issue to persist on your official examples.. I'll look into that.

**Comment by Abdisalan** [comment]
Solved:
The second lock I was using to login a linkedin account was actually also recieving the authenticated event but the resulting token was too large and resulting in an error. I reduced the size of the token by changing the options scope of the lock to `openid` and turning off remember last login feature. 
```javascript
{
    rememberLastLogin: false,
    auth: {
        params: {
            scope: 'openid'
        }
    }
}
```
There are more details in this auth0 community post where I found the solution. [link](https://community.auth0.com/t/the-generated-token-is-too-large/9144)
